jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1114 @@
+
+    def _body_1d(i, args):
+        z, wky, kyind, kxi, nkx, nkx_2, kxp, zp = args
+
+        kxind = (kxi + i) % nkx
+        _kx = kx - (kxp + i)
+        wkx = interp._xval_noraise(_kx)
+
+        val = jnp.where(
+            kxind < nkx_2,
+            zp[(nky - kyind) % nky, nkx - kxind - nkx_2].conjugate(),
+            zp[kyind, kxind - nkx_2],
+        )
+        z += val * wkx * wky
+
+        return [z, wky, kyind, kxi, nkx, nkx_2, kxp, zp]
+
+    def _body(i, args):
+        z, kxi, kyi, nky, nkx, nkx_2, kxp, kyp, zp = args
+        kyind = (kyi + i) % nky
+        _ky = ky - (kyp + i)
+        wky = interp._xval_noraise(_ky)
+        z = jax.lax.fori_loop(
+            -interp.xrange,
+            interp.xrange + 1,
+            _body_1d,
+            [z, wky, kyind, kxi, nkx, nkx_2, kxp, zp],
+        )[0]
+        return [z, kxi, kyi, nky, nkx, nkx_2, kxp, kyp, zp]
+
+    z = jax.lax.fori_loop(
+        -interp.xrange,
+        interp.xrange + 1,
+        _body,
+        [jnp.zeros(kx.shape, dtype=complex), kxi, kyi, nky, nkx, nkx_2, kxp, kyp, zp],
+    )[0]
+    return z.reshape(orig_shape)
+
+
+@jax.jit
+def _flux_frac(a, x, y, cenx, ceny):
+    def _body(d, args):
+        res, a, dx, dy, cenx, ceny = args
+        msk = (jnp.abs(dx) <= d) & (jnp.abs(dx) <= d)
+
+        res = res.at[d].set(
+            jnp.sum(
+                jnp.where(
+                    msk,
+                    a,
+                    0.0,
+                )
+            )
+        )
+
+        return [res, a, dx, dy, cenx, ceny]
+
+    res = jnp.zeros(a.shape[0], dtype=float) - jnp.inf
+    return jax.lax.fori_loop(
+        0, a.shape[0], _body, [res, a, x - cenx, y - ceny, cenx, ceny]
+    )[0]
+
+
+@jax.jit
+def _calculate_size_containing_flux(image, thresh):
+    cenx, ceny = image.center.x, image.center.y
+    x, y = image.get_pixel_centers()
+    fluxes = _flux_frac(image.array, x, y, cenx, ceny)
+    msk = fluxes >= -jnp.inf
+    fluxes = jnp.where(msk, fluxes, jnp.max(fluxes))
+    d = jnp.arange(image.array.shape[0]) + 1.0
+    expfac = 4.0
+    dint = jnp.arange(image.array.shape[0] * expfac) / expfac + 1.0
+    fluxes = jnp.interp(dint, d, fluxes)
+    msk = fluxes <= thresh
+    return (
+        jnp.argmax(
+            jnp.where(
+                msk,
+                dint,
+                -jnp.inf,
+            )
+        )
+        / expfac
+        + 1.0
+    )
+
+
+@jax.jit
+def _inner_comp_find_maxk(arr, thresh, kx, ky):
+    msk = (arr * arr.conjugate()).real > thresh * thresh
+    max_kx = jnp.max(
+        jnp.where(
+            msk,
+            jnp.abs(kx),
+            -jnp.inf,
+        )
+    )
+    max_ky = jnp.max(
+        jnp.where(
+            msk,
+            jnp.abs(ky),
+            -jnp.inf,
+        )
+    )
+    return jnp.maximum(max_kx, max_ky)
+
+
+@jax.jit
+def _find_maxk(kim, max_maxk, thresh):
+    kx, ky = kim.get_pixel_centers()
+    kx *= kim.scale
+    ky *= kim.scale
+    return jnp.minimum(
```

Comment:
why is this a mininum?

Reply:
We are taking the minimum between the maxk we find from the image and an upper bound on maxk, called max_maxk. I added a comment.